src/backend/InvenTree/report/templatetags/barcode.py

```diff
@@ -45,26 +53,31 @@ def clean_barcode(data):
 
 
 @register.simple_tag()
-def qrcode(data, **kwargs):
+def qrcode(data: str, **kwargs) -> str:
     """Return a byte-encoded QR code image.
 
     Arguments:
         data: Data to encode
 
     Keyword Arguments:
-        version: QR code version, (None to auto detect) (default = None)
-        error_correction: Error correction level (L: 7%, M: 15%, Q: 25%, H: 30%) (default = 'M')
-        box_size: pixel dimensions for one black square pixel in the QR code (default = 20)
-        border: count white QR square pixels around the qr code, needed as padding (default = 1)
-        optimize: data will be split into multiple chunks of at least this length using different modes (text, alphanumeric, binary) to optimize the QR code size. Set to `0` to disable. (default = 1)
-        format: Image format (default = 'PNG')
-        fill_color: Fill color (default = "black")
-        back_color: Background color (default = "white")
+        version (int): QR code version, (None to auto detect) (default = None)
+        error_correction (str): Error correction level (L: 7%, M: 15%, Q: 25%, H: 30%) (default = 'M')
+        box_size (int): pixel dimensions for one black square pixel in the QR code (default = 20)
+        border (int): count white QR square pixels around the qr code, needed as padding (default = 1)
+        optimize (int): data will be split into multiple chunks of at least this length using different modes (text, alphanumeric, binary) to optimize the QR code size. Set to `0` to disable. (default = 1)
+        format (str): Image format (default = 'PNG')
+        fill_color (str): Fill color (default = "black")
+        back_color (str): Background color (default = "white")
 
     Returns:
-        base64 encoded image data
+        image (str): base64 encoded image data
 
     """
+    data = str(data).strip()
+
+    if not data:
+        return error_barcode()
```

Comment:
Q: Would it not be better to raise at that point?

Reply:
Good point, this is now implemented